contracts/badger-sett/strategies/unitprotocol/StrategyUnitProtocolMeta.sol

```diff
@@ -0,0 +1,248 @@
+    }
+
+    /// @dev Balance of want currently held in strategy positions
+    function balanceOfPool() public override view returns (uint256) {
+        return getCollateralBalance();
+    }
+
+    function collateralValue(uint256 collateralAmt) public view returns (uint256) {
+        uint256 collateralPrice = getLatestCollateralPrice();
+        return collateralAmt.mul(collateralPrice).mul(1e18).div(collateralDecimal).div(collateralPriceDecimal); // debtToken in 1e18 decimal
+    }
+
+    function currentRatio() public view returns (uint256) {
+        _onlyCDPInUse();
+        uint256 collateralAmt = collateralValue(getCollateralBalance()).mul(100);
+        uint256 debtAmt = getDebtBalance();
+        return collateralAmt.div(debtAmt);
+    }
+
+    // if borrow is true (for addCollateralAndBorrow): return (maxDebt - currentDebt) if positive value, otherwise return 0
+    // if borrow is false (for repayAndRedeemCollateral): return (currentDebt - maxDebt) if positive value, otherwise return 0
+    function calculateDebtFor(uint256 collateralAmt, bool borrow) public view returns (uint256) {
+        uint256 maxDebt = collateralValue(collateralAmt).mul(ratioBuffMax).div(_getBufferedMinRatio(ratioBuffMax));
+
+        uint256 debtAmt = getDebtBalance();
+
+        uint256 debt = 0;
+
+        if (borrow && maxDebt >= debtAmt) {
+            debt = maxDebt.sub(debtAmt);
+        } else if (!borrow && debtAmt >= maxDebt) {
+            debt = debtAmt.sub(maxDebt);
+        }
+
+        return (debt > 0) ? debt : 0;
+    }
+
+    function _getBufferedMinRatio(uint256 _multiplier) internal view returns (uint256) {
+        require(ratioBuffMax > 0, "!ratioBufferMax");
+        require(minRatio > 0, "!minRatio");
+        return minRatio.mul(_multiplier).mul(ratioBuffMax.add(ratioBuff)).div(ratioBuffMax).div(100);
+    }
+
+    function borrowableDebt() public view returns (uint256) {
+        uint256 collateralAmt = getCollateralBalance();
+        return calculateDebtFor(collateralAmt, true);
+    }
+
+    function requiredPaidDebt(uint256 _redeemCollateralAmt) public view returns (uint256) {
+        uint256 collateralAmt = getCollateralBalance().sub(_redeemCollateralAmt);
+        return calculateDebtFor(collateralAmt, false);
+    }
+
+    // **** sub-strategy implementation ****
+    function _depositUSDP(uint256 _usdpAmt) internal virtual;
+
+    function _withdrawUSDP(uint256 _usdpAmt) internal virtual;
+
+    // **** Oracle (using chainlink) ****
+
+    function getLatestCollateralPrice() public view returns (uint256) {
+        require(unitOracle != address(0), "!_collateralOracle");
+
+        (, int256 price, , , ) = IChainlinkAggregator(unitOracle).latestRoundData();
+
+        if (price > 0) {
+            int256 ethPrice = 1;
+            if (collateralPriceEth) {
+                (, ethPrice, , , ) = IChainlinkAggregator(eth_usd).latestRoundData(); // eth price from chainlink in 1e8 decimal
```

Comment:
Can this line be omitted? As we directly have renBTC-USD price feed, so `collateralPriceEth` will remain false for this strategy...
If so, mention of `ethPrice` and `collateralPriceEth ` can also be removed then from the below return statement.